File: tests/D01.cs
namespace tests01
{
    using NUnit.Framework;
    using NFluent;
    using System.IO;
    using src01;

    public class Tests
    {
        [TestCase(12, 2)]
        [TestCase(14, 2)]
        [TestCase(1969, 654)]
        [TestCase(100756, 33583)]
        public void GetFuelForMass(int givenMass, int expectedFuel)
        {
            Check.That(Code.FuelForMass(givenMass)).IsEqualTo(expectedFuel);
        }
        
        [Test]
        public void GetSumOfFuelForMasses()
        {
            Check.That(
                new string[] {"12", "14", "1969"}.SumOf(Code.FuelForMass)
            ).IsEqualTo(658);
        }
        
        [Test]
        public void Part1()
        {
            Check.That(
                File.ReadAllLines("D01.txt").SumOf(Code.FuelForMass)
            ).IsEqualTo(3401852);
        }

        [TestCase(12, 2)]
        [TestCase(14, 2)]
        [TestCase(1969, 966)]
        [TestCase(100756, 50346)]
        public void GetTotalFuelForMass(int givenMass, int expectedFuel)
        {
            Check.That(Code.TotalFuelForMass(givenMass)).IsEqualTo(expectedFuel);
        }
        
        [Test]
        public void GetSumOfTotalFuelForMasses()
        {
            Check.That(
                new string[] {"12", "14", "1969"}.SumOf(Code.TotalFuelForMass)
            ).IsEqualTo(970);
        }
        
        [Test]
        public void Part2()
        {
            Check.That(
                File.ReadAllLines("D01.txt").SumOf(Code.TotalFuelForMass)
            ).IsEqualTo(5099916);
        }
    }
}

File: testsf/D01.fs
namespace Tests

open System.IO
open NUnit.Framework
open NFluent
open Src01

[<TestFixture>]
type Tests01() =

    [<TestCase(12, 2)>]
    [<TestCase(14, 2)>]
    [<TestCase(1969, 654)>]
    [<TestCase(100756, 33583)>]
    member this.GetFuelForMass(givenMass,expectedFuel) =
        Check.That(FuelForMass givenMass).IsEqualTo(expectedFuel) |> ignore
        
    [<Test>]
    member this.GetSumOfFuelForMasses() =
        Check.That(["12"; "14"; "1969"] |> SumOf FuelForMass).IsEqualTo(658) |> ignore    

    [<Test>]
    member this.Part1() =
        Check.That(this.MyInput |> SumOf FuelForMass).IsEqualTo(3401852) |> ignore

    [<TestCase(12, 2)>]
    [<TestCase(14, 2)>]
    [<TestCase(1969, 966)>]
    [<TestCase(100756, 50346)>]
    member this.GetTotalFuelForMass(givenMass,expectedFuel) =
        Check.That(TotalFuelForMass givenMass).IsEqualTo(expectedFuel) |> ignore
        
    [<Test>]
    member this.GetSumOfTotalFuelForMasses() =
        Check.That(["12"; "14"; "1969"] |> SumOf TotalFuelForMass).IsEqualTo(970) |> ignore    

    [<Test>]
    member this.Part2() =
        Check.That(this.MyInput |> SumOf TotalFuelForMass).IsEqualTo(5099916) |> ignore

    member this.MyInput = File.ReadAllLines "D01.txt"
